baseplate/server/__init__.py

```diff
@@ -226,7 +227,7 @@ def make_app(app_config: Dict[str, str]) -> Callable:
 
 
 def register_signal_handlers() -> threading.Event:
-    def _handle_debug_signal(_signo: int, frame: FrameType) -> None:
+    def _handle_debug_signal(_signo: int, frame: FrameType | None) -> None:
```

Comment:
🔕 Why not `Optional[FrameType]`? Just curious.

Reply:
I personally prefer the semantics of `Optional[X]` over `X | None` (I think it suggests intent more clearly) but my impression is that the the latter has "won" the general consensus and is considered more idiomatic. For example, [pyupgrade](https://github.com/asottile/pyupgrade) automatically converts type annotations to it. I tried to find a more official source but opinions are kind of mixed: https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945

Happy to change it back if you prefer. They should be functionally equivalent.